common.js

```diff
@@ -0,0 +1,142 @@
+}
+exports.fn_info = async function (client, fn_name, nmspace) {
+    const fn_info = await client.apis['fission.io'].v1.namespaces(nmspace).functions(fn_name).getStream();
+    console.log(fn_info);
+}
+
+ exports.create_env = async function (client,env_name,nmspace,img) {
+    const env = {
+        apiVersion: 'fission.io/v1',
+        kind: 'Environment',
+        metadata: {
+            name: env_name,
+            namespace: nmspace,
+        },
+        spec: {
+            TerminationGracePeriod: 360,
+            builder: {},
+            keeparchive: false,
+            poolsize: 1,
+            resources: {},
+            runtime: {
+                functionendpointport: 0,
+                image: img,
+                loadendpointpath: '',
+                loadendpointport: 0
+            },
+        version: 1
+        }
+    }
+    const envs = await client.apis['fission.io'].v1.namespaces('default').environments.post({body: env});
+    console.log("Fission environment ",envs['body']['metadata']['name']," created");
+}
+
+async function create_func(client,funcname,env_name,pkg_name) {
+
+    const fn = {
+        apiVersion: 'fission.io/v1',
+        kind: 'Function',
+        metadata: {
+            clusterName: '',
+            name: funcname,
+            namespace: 'default',
+        },
+        spec: {
+            InvokeStrategy: {
+                ExecutionStrategy: ['poolmgr'],
+                StrategyType: 'execution'
+            },
+            configmaps: null,
+            environment: {
+                name: env_name,
+                namespace: 'default',
+            },
+            package: {
+                packageref: {
+                    namespace: 'default',
+                    name: pkg_name['metadata']['name'],
+                    resourceVersion: pkg_name['metadata']['resourceVersion'],
+       }
+            },
+            resources: {
+            },
+            secrets: null
+        }
+    }
+    const fns = await client.apis['fission.io'].v1.namespaces('default').functions.post({body: fn});
+    console.log("Fission function ", fns['body']['metadata']['name'], " created");
+}
+
+async function create_pkgs(client, archive) {
+    const pkg = {
+        apiVersion: 'fission.io/v1',
+        kind: 'Package',
+        metadata: {
+            clusterName: '',
+            name: 'test-js-yc0r',
+            namespace: 'default'
+        },
+        spec: {
+            deployment: archive,
+            environment: {
+                name: 'node',
+                namespace: 'default'
+            },
+           source: {
+                checksum: {}
+            }
+        }
+    }
+    const pkgs = await client.apis['fission.io'].v1.namespaces('default').packages.post({ body: pkg});
+    return pkgs;
+}
+
+function get_contents(filePath, cb) {
+    fs.readFile(filePath, {encoding: 'utf-8'}, function(err,data){
+        if (err) {
+            console.log(err);
+        }
+        cb(data);
```

Comment:
shouldn't this be in else statement ?

Reply:
no because js and fission show the output of whatever is compiled to the point of error which make it easier to track the error